File: Core/Ecommerce.Domain/Entities/Review.cs
using System.ComponentModel.DataAnnotations;
using Ecommerce.Domain.Common;
// ReSharper disable EntityFramework.ModelValidation.UnlimitedStringLength

namespace Ecommerce.Domain.Entities
{
	public class Review : AuditableEntity
	{
		public int Id { get; set; }
		
		public int ProductId { get; set; }
		
		[MaxLength(256)]
		public string UserName { get; set; } = null!;
		
		public int Stars { get; set; }
		
		public string? Comments { get; set; }
	}
}

File: UI/ecommerce-ui/src/models/Review.ts
export class Review {
    id: number;
    productId: number;
    userName: string;
    stars: number;
    comments: string;

    constructor(
        id: number,
        productId: number,
        userName: string,
        stars: number,
        comments: string
    ) {
        this.id = id;
        this.productId = productId;
        this.userName = userName;
        this.stars = stars;
        this.comments = comments;
    }
}
